EZEE-3357 - Migration exception fix (#1703)
ezsystems/developer-documentation: docs/migrating/migrating_from_ez_publish.md

- Rewrite custom admin modules to use the new Platform/Symfony stack
    - And if you do this while on 5.x, you can use several of the [available legacy migration features](https://doc.ez.no/display/EZP/Legacy+code+and+features) to make the new code appear in legacy admin

See Upgrade documentation on how to perform the actual upgrade: [Upgrade (eZ Publish Platform page)](https://doc.ez.no/display/EZP/Upgrade).

!!! caution "Avoid exception when migrating the database"

    If you plan to migrate from from eZ Publish through eZ Publish Platform 5.4 to eZ Platform and further, an exception may occur when you try to migrate the database while it contains internal drafts of Landing Pages.

>     This can happen because such drafts do not have an expected row in the `ezcontentobject_name` table.<a id="migration_exception"></a> 

Reviewer: what does this `<a id="migration_exception"></a>` do? Anchors should usually have some content inside them so they're visible/clickable.

Author: No need to be either visible or clickable - it is merely a trick to help redirect from the other occurrence of this warning to the detailed recommendations.

    
    To avoid this exception, you must remove all internal drafts before you migrate. 
    First, in `content.ini`, set the `InternalDraftsCleanUpLimit` and `InternalDraftsDuration` values to 0. 

>     Then [run](https://symfony.com/doc/current/the-fast-track/en/24-cron.html) the [internal drafts cleanup](https://github.com/ezsystems/ezpublish-legacy/blob/2019.03/cronjobs/internal_drafts_cleanup.php) cron job. 

Reviewer: You don't need cron to run a script one time. Simply state `Then run the [internal drafts cleanup](...)` script.

    - And if you do this while on 5.x, you can use several of the [available legacy migration features](https://doc.ez.no/display/EZP/Legacy+code+and+features) to make the new code appear in legacy admin

> See Upgrade documentation on how to perform the actual upgrade: [Upgrade (eZ Publish Platform page)](https://doc.ez.no/display/EZP/Upgrade)

Reviewer 2: Something is wrong with rendering and this line does now show in the output at all.

See Upgrade documentation on how to perform the actual upgrade: [Upgrade (eZ Publish Platform page)](https://doc.ez.no/display/EZP/Upgrade)

!!! caution "Avoid exception when migrating the database"

>     If you plan to migrate from from eZ Publish through eZ Publish Platform 5.4 to eZ Platform and further, an exception may occur when you try to migrate the database while it contains internal drafts of Landing Pages, because such drafts do not have an expected row in the `ezcontentobject_name` table. 

Reviewer 2: ```suggestion
    If you migrate from from eZ Publish through eZ Publish Platform 5.4 to eZ Platform and further, an exception may occur when you try to migrate the database if it contains internal drafts of Landing Pages.
    This can happen because such drafts do not have an expected row in the `ezcontentobject_name` table. 
```
an attempt to shorten a pretty long sentence

>     To overcome this obstacle, you must first remove all internal drafts by setting the `InternalDraftsCleanUpLimit` and `InternalDraftsDuration` values to 0 and [running](https://doc.ez.no/eZ-Publish/Technical-manual/4.x/Features/Cronjobs/Running-cronjobs) the [internal drafts cleanup](https://github.com/ezsystems/ezpublish-legacy/blob/2019.03/cronjobs/internal_drafts_cleanup.php) cron job. 

Reviewer 2: ```suggestion
    To avoid this exception, you must first remove all internal drafts by setting the `InternalDraftsCleanUpLimit` and `InternalDraftsDuration` values to 0 and [running](https://doc.ez.no/eZ-Publish/Technical-manual/4.x/Features/Cronjobs/Running-cronjobs) the [internal drafts cleanup](https://github.com/ezsystems/ezpublish-legacy/blob/2019.03/cronjobs/internal_drafts_cleanup.php) cron job. 
```
The old 4.x doc is planned to be disabled soon-ish. Can we handle it somehow without linking to it?
